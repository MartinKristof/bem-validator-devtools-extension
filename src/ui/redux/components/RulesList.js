import React from 'react';
import PropTypes from 'prop-types';
import uuid from 'uuid';
import RulesItem from './RulesItem';

const RulesList = ({ rules, inspectElement }) =>
  rules.map((rule) => <RulesItem key={uuid()} rule={rule} inspectElement={inspectElement} />);

RulesList.propTypes = {
  rules: PropTypes.arrayOf(
    PropTypes.exact({ missingClassName: PropTypes.string.isRequired, selector: PropTypes.string.isRequired }),
  ).isRequired,
  inspectElement: PropTypes.func.isRequired,
};

export default RulesList;
