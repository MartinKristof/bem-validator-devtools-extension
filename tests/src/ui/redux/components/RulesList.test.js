import React from 'react';
import { shallow } from 'enzyme';
import RulesList from '../../../../../src/ui/redux/components/RulesList';

jest.mock('uuid', () => {
  let value = 0;

  return () => {
    value += 1;

    return value;
  };
});

describe('<RulesList />', () => {
  describe('#render', () => {
    it('should render without side-effects', () => {
      const rules = [{ missingClassName: '.foo', selector: '.bar' }, { missingClassName: '.boo', selector: '.bazz' }];
      const node = shallow(<RulesList inspectElement={() => {}} rules={rules} />);

      expect(node).toMatchSnapshot();
    });
  });
});
