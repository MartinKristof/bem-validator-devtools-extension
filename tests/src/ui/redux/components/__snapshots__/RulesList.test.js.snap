// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RulesList /> #render should render without side-effects 1`] = `
Array [
  <RulesItem
    inspectElement={[Function]}
    key="1"
    rule={
      Object {
        "missingClassName": ".foo",
        "selector": ".bar",
      }
    }
  />,
  <RulesItem
    inspectElement={[Function]}
    key="2"
    rule={
      Object {
        "missingClassName": ".boo",
        "selector": ".bazz",
      }
    }
  />,
]
`;
